<launch>
  <!-- rosbag play -->

  <param name="use_sim_time" value="true"/>
  <!--<arg name="bag"/> 
  
  <node pkg="rosbag" type="play" name="rosbag_play" args="clock $(arg bag)" required="false"/>-->

  <!--<node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 /velodyne /odom 100"/> -->
  
  <node pkg="velodyne_pointcloud" type="cloud_node" name="cloud_node">
    <param name="calibration" value="$(find object3d_detector)/config/vlp16.yaml"/>
    <param name="min_range" value="0.4"/>
    <param name="max_range" value="130.0"/>
  </node>
  
  <!-- Velodyne VLP-16 Driver
  <node pkg="nodelet" type="nodelet" name="velodyne_nodelet_manager" args="manager" output="screen"/>
  
  <node pkg="nodelet" type="nodelet" name="driver_nodelet" args="load velodyne_driver/DriverNodelet velodyne_nodelet_manager" output="screen">
    <param name="model" value="VLP16"/>
    <param name="pcap" value=""/>
    <param name="frame_id" value="velodyne"/>
    <param name="rpm" value="600.0"/>
  </node>
  
  <node pkg="nodelet" type="nodelet" name="cloud_nodelet" args="load velodyne_pointcloud/CloudNodelet velodyne_nodelet_manager" output="screen">
    <param name="calibration" value="$(find object3d_detector)/config/vlp16.yaml"/>
    <param name="min_range" value="0.4"/>
    <param name="max_range" value="130.0"/>
  </node>
  -->
  
  <!-- FLOBOT 3D Object Detector -->
  <node pkg="adaptive_clustering" name="adaptive_clustering" type="adaptive_clustering">
    <param name="sensor_model" value="VLP-16"/>
    <param name="obj_size_limit" type="bool" value="true"/>
    <param name="z_axis_min" value="-0.8"/>
    <param name="cluster_size_max" value="5000"/>
    <param name="print_fps" value="true"/>
  </node>
  
  <!-- NBellotto's Bayes People Tracker -->
  <rosparam command="load" file="$(find adaptive_clustering)/config/object3d_detector.yaml"/>
  <node pkg="bayes_people_tracker" type="bayes_people_tracker" name="bayes_people_tracker" output="screen" respawn="false">
    <param name="target_frame" type="string" value="world"/>
  </node>
  
  <!-- ROS Visualization -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find adaptive_clustering)/launch/adaptive_clustering.rviz"/>
</launch>
